wildwood/forest.py

```diff
@@ -1018,18 +1079,41 @@ def _encode_y(self, y):
         output : ndarray
             Encoded array of labels
         """
-        #
-        # and n_trees_per_iteration_
+        multiclass = self.multiclass
         check_classification_targets(y)
-        label_encoder = LabelEncoder()
-        encoded_y = label_encoder.fit_transform(y)
+        # One-hot encode the labels for "ovr", otherwise just use ordinal encoding
+        label_encoder = LabelBinarizer() if multiclass == "ovr" else LabelEncoder()
+        pre_encoded_y = label_encoder.fit_transform(y)
         self._classes_ = label_encoder.classes_
         n_classes_ = self._classes_.shape[0]
         self._n_classes_ = n_classes_
         # only 1 tree for binary classification.
         # TODO: For multiclass classification, we build 1 tree per class.
         # self.n_trees_per_iteration_ = 1 if n_classes_ <= 2 else n_classes_
-        encoded_y = np.ascontiguousarray(encoded_y, dtype=np.float32)
+
+        if multiclass == "ovr" and n_classes_ <= 2:
+            if self.verbose:
+                warn(
+                    "Only two classes where detected: switching to "
+                    'multiclass="multiclass" instead of "ovr"'
+                )
+            multiclass = "multinomial"
+            self.multiclass = multiclass
+
+        if multiclass == "ovr":
+            # TODO: pourquoi une liste de arrays et numpy array 2D ?
+            encoded_y = [
+                np.ascontiguousarray(pre_encoded_y[:, i], dtype=np.float32)
+                for i in range(n_classes_)
+            ]
+        else:
+            encoded_y = np.ascontiguousarray(pre_encoded_y, dtype=np.float32)
+            # if multiclass == "multinomial":
+            #
+            # else:
+            #     encoded_y = np.ascontiguousarray(
+            #         pre_encoded_y.flatten(), dtype=np.float32
+            #     )
         return encoded_y
```

Comment:
The flatten is necessary here , if the classifier is called with multiclass="ovr" and the classification turns out to be binary, there will be a numba compilation error here because the encoded_y will have a dimension too many here. we need to keep the initial value of multiclass and use it in this if else